src/util/gas-factory-helpers.ts

```diff
@@ -250,12 +250,8 @@ export async function calculateGasUsed(
   const gasPriceWei = route.gasPriceWei;
   // calculate L2 to L1 security fee if relevant
   let l2toL1FeeInWei = BigNumber.from(0);
-  if ([ChainId.ARBITRUM_ONE, ChainId.ARBITRUM_GOERLI].includes(chainId)) {
-    l2toL1FeeInWei = calculateArbitrumToL1FeeFromCalldata(
-      route.methodParameters!.calldata,
-      l2GasData as ArbitrumGasData
-    )[1];
-  } else if (
```

Comment:
I actually think there's no double counting of L1 fees here, although the reason is a bit complex.

You probably noticed two codepaths that invokes `calculateGasUsed`:

1. [EthEstimateGasSimulator.ethEstimateGas](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/eth-estimate-gas-provider.ts#L117)
2. [TenderlySimulator.simulateTransaction](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/tenderly-simulation-provider.ts#L527)

Those two codepaths are actually mutually exclusive to each other, due to the check in [FallbackTenderlySimulator.simulateTransaction](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/tenderly-simulation-provider.ts#L126C7-L131C9). Then inside [EthEstimateGasSimulator.ethEstimateGas](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/eth-estimate-gas-provider.ts#L117), from Arbitrum [doc](https://docs.arbitrum.io/devs-how-tos/how-to-estimate-gas), I see the node returns the entire transaction fee at the specified L2 gas price. I assume it does not include the L1 calldata posting costs. For Tenderly simulation, I'm a bit less certain if they include the L1 fee as well.

Taking a step back, the change inside `calculateGasUsed` will likely have no effect, if your experience is such that you notice the abnormally high gas cost using the Uniswap web interface, not mobile. The reason is a bit complex.

`calculateGasUsed`  is only going to be invoked when upstream requests smart order router to get a best swap route with forked net simulation. The code is inside [AlphaRouter](https://github.com/Uniswap/smart-order-router/blob/main/src/routers/alpha-router/alpha-router.ts#L1498). Majority of the Arbitrum abnormal high gas cost report I see is the web app screenshot, so it means the Arbitrum gas cost is not recomputed inside gas-factory-helpers, but only once inside [V3HeuristicGasModelFactory](https://github.com/Uniswap/smart-order-router/blob/main/src/routers/alpha-router/gas-models/v3/v3-heuristic-gas-model.ts#L417)

Reply:
> I assume it does not include the L1 calldata posting costs. 
This is incorrect, unlike Optimism, Arbitrum charge `L2 gas` for `L1 calldata posting costs`. For example, a normal transfer cost 21000 L2 gas to execute and say 100000 L2 gas for L1 calldata, Arbitrum require the user to sign a transaction with at least 121000 gas to be included onchain. The RPC method "eth_estimateGas" on an Arbitrum node return the TOTAL L2 gas required, that is `L2 execution gas + L1 calldata gas` which for the above example would be 121000. As far as I know, tenderly also have the same behavior (that the gas estimate include L1 calldata gas)

Reply:
In fact I don't really see why we need `calculateArbitrumToL1FeeFromCalldata` anywhere except for display purpose (show user L1 and L2 fee split) because the gas estimation would have returned the total already.

Reply:
> This is incorrect, unlike Optimism, Arbitrum charge L2 gas for L1 calldata posting costs. 

This is good to know. This assumption is embedded in sor gas model. Let me think through how to fix this assumption.

Reply:
I am running the cli quoter, `estimatedGasUsed` seems to be wrong but I cannot figure out where it come from. If it uses eth_estimateGas it should return a much higher number.

Reply:
Ah it does not do eth_estimateGas, it was [semi-hardcoded in the gas model](https://github.com/gzeoneth/smart-order-router/blob/ad63c7fb36c9cbc12a543681124f0a4afa53f642/src/routers/alpha-router/gas-models/v3/v3-heuristic-gas-model.ts#L326-L327)

Reply:
I think I am not familiar with the SOR to properly fix this, but would be nice if we can ship the div(16) fix first to make the number at least much more reasonable. Thanks for the quick response btw.

Reply:
Agree, there's a bit of urgency. Let's ship this tomorrow during EST/PST business hours. I can help create a replicated PR within
[https://github.com/Uniswap/smart-order-router](https://github.com/Uniswap/smart-order-router), so all the tests should pass.

Reply:
Cool, feel free to cherrypick my commit

Reply:
https://github.com/Uniswap/smart-order-router/pull/466 replicated here